### PR TITLE
Schedule preorder ending

### PR DESCRIPTION
**Pull request by Kwaidan00**
I want to merge this change because it uses `django_celery_beat` to dynamically add tasks and schedules to run task for ending preorders at the correct time.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2998 Preorders](https://app.clickup.com/t/2549495/SALEOR-2998)

**Comment by patrys** [comment]
Task linked: [SALEOR-3808 Implement scheduling task for preorder ending](https://app.clickup.com/t/2549495/SALEOR-3808)

File: saleor/product/tasks.py [hunk 1]
```diff
@@ -85,3 +90,116 @@ def update_products_discounted_prices_of_discount_task(discount_pk: int):
...
+        "minute": str(end_time.minute),
+        "hour": str(end_time.hour),
+        "day_of_week": "*",
+        "day_of_month": str(end_date.day),
+        "month_of_year": str(end_date.month),
+        "timezone": None,
+    }
+
+
+def deactivate_preorder_for_variant_task(product_variant_id: int):
```
Comment (fowczarek) [inline saleor/product/tasks.py:169]: This function is not called?

**Review by fowczarek — COMMENTED** [review]
1. Should we call `delete_deactivate_preorder_for_variant_task`  when product Variant is deleted?
2. There is a way to change the preorder end date? If yes, do we update tasks in such a situation?

File: saleor/product/tasks.py (hunk 1 above)
Comment (Kwaidan00) [reply saleor/product/tasks.py:169]: It's the Celery task, I forgot about decorator.

**Comment by Kwaidan00** [comment]
Closed in favor of https://github.com/mirumee/saleor/pull/8234

**Pull request closed without merging**